src/odemis/gui/comp/viewport.py

```diff
@@ -393,27 +393,30 @@ def _checkMergeSliderDisplay(self):
                 # => it should be done in the MicroscopeView when adding a stream
                 # For now, special hack for the MicroscopeCanvas which always sets
                 # the EM image as "right" (ie, it's drawn last).
-                # If there is SEM and Spectrum, the spectrum image is always
+                # If there is EM and Spectrum or CL, the spectrum/CL image is always
                 # set as "right" (ie, it's drawn last).
+                # Note: in practice, there is no AR spatial stream, so it's
+                # never mixed with any other stream.
                 if (
-                        any(isinstance(s, EMStream) for s in streams)
-                        and any(isinstance(s, OpticalStream) for s in streams)
+                        any(isinstance(s, EMStream) for s in streams) and
+                        any(isinstance(s, OpticalStream) for s in streams)
                 ):
-                    self.bottom_legend.set_stream_type(wx.LEFT, model.MD_AT_CL)  # CL and Fluo
-                    # types correspond to same icon
+                    # TODO: don't hard-code MD_AT_FLUO and MD_AT_EM but use the
+                    # acquisitionType of the corresponding streams.
+                    self.bottom_legend.set_stream_type(wx.LEFT, model.MD_AT_FLUO)
                     self.bottom_legend.set_stream_type(wx.RIGHT, model.MD_AT_EM)
                 elif (
-                        any(isinstance(s, EMStream) for s in streams)
-                        and any(isinstance(s, SpectrumStream) for s in streams)
+                        any(isinstance(s, EMStream) for s in streams) and
+                        any(isinstance(s, SpectrumStream) for s in streams)
                 ):
                     self.bottom_legend.set_stream_type(wx.LEFT, model.MD_AT_EM)
                     self.bottom_legend.set_stream_type(wx.RIGHT, model.MD_AT_SPECTRUM)
                 elif (
-                        any(isinstance(s, EMStream) for s in streams)
-                        and any(isinstance(s, ARStream) for s in streams)
+                        any(isinstance(s, EMStream) for s in streams) and
+                        any(isinstance(s, CLStream) for s in streams)
                 ):
-                    self.bottom_legend.set_stream_type(wx.LEFT, model.MD_AT_AR)
```

Comment:
It is possible to show AR and SEM together in the tiling window. In this case the icons are at the wrong side.